### ISSUE TITLE
Unusable button show for "swipe"-ability

### ISSUE THREAD

**Issue by NoNamePaul** [brittyazel/Neuron#102]
Usually if a button is assign to an ability that is not usable in the current state (e.g. when you have not specced into it) it's just an transparent button and nothing happens when you click it.
This works fine except for one specific ability:
The ability [swipe](http://www.wowhead.com/spell=106785/swipe) from the talent [Feral Affinity](http://www.wowhead.com/spell=202157/feral-affinity) just ignores the fact that its not usable. (tested with balance and restoration spec, same result)
All other abilities from all other affinity talents disappear correctly when I'm specced into a different one.
The button for swipe though stays and trying to click it results in the error message "Spell not learned".
And it frequently is super delayed in loading the icon. e.G. when i go into cat form it sometimes takes like 2 seconds or more for the icon to appear (never had this issue with any other button, and it works fine when I'm specced into feral affinity)

**Comment by brittyazel** [comment]
Yeah, this is probably due to a hack I had to put in for the swipe ability. The issue was primarily with the fact that there are like 6 swipe spellID's and for some reason Blizzard's code is somewhat broken. In some instances the cat for swipe will get loaded up in bear form and visa versa. Also, the ability ID that registers a "proc" for swipe is different from the ability ID that you use to actually use the ability. So I had to add extra checking in there to account for that. These were all hacks though to just get feral and guardian druids playable, so maybe I'll revist this as well and see what I can do

**Comment by NoNamePaul** [comment]
If I have time this weekend I will look into this myself, since it probably bugs me the most and you labled it with "help wanted" :D 
And I'm usually quite good with finding solutions for such problems and workarounds for a bad API ^^
I'll let you know should i be able to fix it.

**Comment by brittyazel** [comment]
awesome, thanks for the help. I just tested this myself and added swipe to a bar when in feral spec, and then switched to resto. My swipe ability correctly disappeared from the bar

**Comment by brittyazel** [comment]
I also tested in like you did. In resto spec > activate feral affinity > add swipe to a bar > switch back to guardian affinity > the swipe button dissapears.

Seems to be working properly....hmm...

**Comment by NoNamePaul** [comment]
I'll check again when I get home later

**Comment by NoNamePaul** [comment]
So i checked again and it's still the same for me (tried with new bars, etc)
It isn't really bad, so don't bother about ist :D I'm just picky there :P 
If I somehow find the time to have a look into it I will, but as of right now i probably won't be able to do that :/
